.semaphore/vms/create-test-vms

```diff
@@ -16,27 +16,54 @@
 
 set -e
 
-source $(pwd)/.semaphore/batches.sh
-
-vm_name_prefix=$1
-project=unique-caldron-775
-zone=${ZONE:-europe-west3-c}
 my_dir="$(dirname $0)"
 repo_dir="$my_dir/../.."
+
+# shellcheck source=../../felix/.semaphore/batches.sh
+source "$repo_dir/$COMPONENT/.semaphore/batches.sh"
+
+vm_name_prefix=$1
 artifacts_dir="$repo_dir/artifacts"
+zone=${ZONE:-europe-west3-c}
+: "${IMAGE_FAMILY:=ubuntu-minimal-2204-lts}"
+: "${MAX_RUN_DURATION:=4h}"
+disk_size=${VM_DISK_SIZE:-20GB}
 
 pids=()
+names=""
+for batch in "${batches[@]}"; do
+  vm_name="$vm_name_prefix$batch"
+  if [ -n "$names" ]; then
+    names="$names,$vm_name"
+  else
+    names="$vm_name"
+  fi
+done
+
+# Do a bulk create; this is faster and it saves API quota.
+gcloud --quiet compute instances bulk create \
+       --predefined-names="$names" \
+       --zone=${zone} \
+       --machine-type=n4-standard-4 \
+       --image-family=${IMAGE_FAMILY} \
+       --image-project=ubuntu-os-cloud \
+       --boot-disk-size=$disk_size \
+       --boot-disk-type=hyperdisk-balanced \
+       --max-run-duration="${MAX_RUN_DURATION}" \
+       --instance-termination-action=DELETE \
+       --metadata-from-file startup-script="$my_dir/vm-bootstrap.sh" \
+       --metadata block-project-ssh-keys=TRUE,ssh-keys="ubuntu:$(ssh-keygen -y -f $HOME/.ssh/id_rsa)",enable-guest-attributes=TRUE
```

Comment:
Do we have a quota on the number of VMs that can be created at a time or is it just the API quota? Is this bulk create `all or nothing` way to create VMs?